src/lib/controllib/BlockGyroCorrected.hpp

```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2017 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include "block/Block.hpp"
+
+#include "BlockBoardRotation.hpp"
+#include <matrix/math.hpp>
+#include <mathlib/mathlib.h>
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/sensor_bias.h>
+#include <uORB/topics/sensor_correction.h>
+#include <uORB/topics/sensor_gyro.h>
+
+namespace control
+{
+
+static constexpr int MAX_GYRO_COUNT = 3;
+
+class __EXPORT BlockGyroCorrected : public SuperBlock
+{
+public:
+
+	BlockGyroCorrected(SuperBlock *parent = nullptr);
+	~BlockGyroCorrected();
+
+	// no copy, assignment, move, move assignment
+	BlockGyroCorrected(const BlockGyroCorrected &) = delete;
+	BlockGyroCorrected &operator=(const BlockGyroCorrected &) = delete;
+	BlockGyroCorrected(BlockGyroCorrected &&) = delete;
+	BlockGyroCorrected &operator=(BlockGyroCorrected &&) = delete;
+
+	const matrix::Vector3f &get() { return _rates; }
+	const uint64_t &timestamp() { return _timestamp; }
+
+	bool poll(int timeout = 100);
```

Comment:
missing documentation:
- unit of timestamp
- it's not obvious that this will call `update()`

Reply:
I was thinking of changing it update blocking or something rather than overloading something called poll.

Reply:
That's fine with me. It's mostly about it being easy to understand & use